tensorboard/main_dev.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""TensorBoard dev main module.
+
+This module configures TensorBoard program for development. For the production
+configuration, please refer to main.py.
+"""
+
+import os
+
+# TF versions prior to 1.15.0 included default GCS filesystem caching logic
+# that interacted pathologically with the pattern of reads used by TensorBoard
+# for logdirs. See: https://github.com/tensorflow/tensorboard/issues/1225
+# The problematic behavior was fixed in 1.15.0 by
+# https://github.com/tensorflow/tensorflow/commit/e43b94649d3e1ac5d538e4eca9166b899511d681
+# but for older versions of TF, we avoid a regression by setting this env var to
+# disable the cache, which must be done before the first import of tensorflow.
+os.environ["GCS_READ_CACHE_DISABLED"] = "1"
+
+from absl import app
+
+# Import this to check that app.run() will accept the flags_parser argument.
+from absl.flags import argparse_flags  # noqa: F401
+
+import inspect
+import sys
+
+from tensorboard import default
+from tensorboard import program
+from tensorboard.compat import tf
+from tensorboard.uploader import uploader_subcommand
+from tensorboard.util import tb_logging
+
+
+logger = tb_logging.get_logger()
+
+
+def run_main():
+    """Initializes flags and calls main()."""
+    logger.info("TensorBoard running in development mode.")
+
+    program.setup_environment()
+
+    if getattr(tf, "__version__", "stub") == "stub":
+        print(
+            "TensorFlow installation not found - running with reduced feature set.",
+            file=sys.stderr,
+        )
+
+    path = os.path.join(
+        os.path.dirname(inspect.getfile(sys._getframe(1))), "dev_webfiles.zip"
+    )
+
+    tensorboard = program.TensorBoard(
+        default.get_plugins(),
+        lambda: open(path, "rb"),
+        subcommands=[uploader_subcommand.UploaderSubcommand()],
+    )
+
+    app.run(tensorboard.main, flags_parser=tensorboard.configure)
```

Comment:
Can we preserve the handling for FlagsError?  Nice to not have extra behavior differences between dev and non-dev versions, and it's only a few lines.